tests/src/App/MappedElement.cpp

```diff
@@ -0,0 +1,143 @@
+    auto originalIndex = originalMappedElement.index;
+
+    // Act
+    Data::MappedElement newMappedElement {std::move(originalMappedElement)};
+
+    // Assert
+    EXPECT_EQ(originalName, newMappedElement.name);
+    EXPECT_EQ(originalIndex, newMappedElement.index);
+}
+
+TEST_F(MappedElementTest, assignmentOperator)
+{
+    // Arrange
+    auto mappedElementA = givenMappedElement("EDGE1", "OTHER_NAME");
+    auto mappedElementB = givenMappedElement("EDGE2", "ANOTHER_NAME");
+    EXPECT_NE(mappedElementA, mappedElementB);// Verify test setup
+
+    // Act
+    mappedElementA = mappedElementB;
+
+    // Assert
+    EXPECT_EQ(mappedElementA, mappedElementB);
+}
+
+TEST_F(MappedElementTest, moveAssignmentOperator)
+{
+    // Arrange
+    auto mappedElementA = givenMappedElement("EDGE1", "OTHER_NAME");
+    auto mappedElementB = givenMappedElement("EDGE2", "ANOTHER_NAME");
+    EXPECT_NE(mappedElementA, mappedElementB);// Verify test setup
+
+    // Act
+    mappedElementA = std::move(mappedElementB);
+
+    // Assert
+    EXPECT_EQ(mappedElementA.name.toString(), "ANOTHER_NAME");
+    EXPECT_EQ(mappedElementA.index.toString(), "EDGE2");
+}
+
+TEST_F(MappedElementTest, equalityOperatorsWhenNotEqual)
+{
+    // Arrange
+    auto mappedElementA = givenMappedElement("EDGE1", "OTHER_NAME");
+    auto mappedElementB = givenMappedElement("EDGE2", "ANOTHER_NAME");
+
+    // Act
+    bool aAndBAreEqual = mappedElementA == mappedElementB;
+    bool aAndBAreNotEqual = mappedElementA != mappedElementB;
+
+    // Assert
+    EXPECT_TRUE(aAndBAreNotEqual);
+    EXPECT_FALSE(aAndBAreEqual);
+}
+
+TEST_F(MappedElementTest, equalityOperatorsWhenEqual)
+{
+    // Arrange
+    auto mappedElementA = givenMappedElement("EDGE1", "OTHER_NAME");
+    auto mappedElementB = givenMappedElement("EDGE1", "OTHER_NAME");
+
+    // Act
+    bool aAndBAreEqual = mappedElementA == mappedElementB;
+    bool aAndBAreNotEqual = mappedElementA != mappedElementB;
+
+    // Assert
+    EXPECT_FALSE(aAndBAreNotEqual);
+    EXPECT_TRUE(aAndBAreEqual);
+}
+
+TEST_F(MappedElementTest, lessThanOperator)
+{
+    // Index is compared first, then mappedName
+
+    // Arrange
+    std::vector<Data::IndexedName> indexedNames {
+        Data::IndexedName {"EDGE1"},
+        Data::IndexedName {"EDGE2"},
+        Data::IndexedName {"EDGE3"},
+    };
+    std::vector<Data::MappedName> mappedNames {
+        Data::MappedName {"A"},
+        Data::MappedName {"B"},
+        Data::MappedName {"C"},
+    };
+    std::vector<Data::MappedElement> mappedElements;
+    // Create a vector of mappedElements that is sorted lowest to highest:
+    for (const auto& indexedName : indexedNames) {
+        for (const auto& mappedName : mappedNames) {
+            mappedElements.emplace_back(indexedName, mappedName);
+        }
+    }
+
+    // Act & Assert
+    for (size_t i = 0; i < mappedElements.size() - 1; ++i) {
+        EXPECT_TRUE(mappedElements[i] < mappedElements[i + 1]);
+    }
```

Comment:
Looks like overkill. Also, do we need to confirm equal mapped elements return false?

Reply:
Yeah, I could certainly simplify this to just a couple of predefined comparisons, and including the false comparison when equal is a good idea. I'll rewrite it.

Reply:
Also, false for `mappedElements[i + 1] < mappedElements[i]`.

Reply:
Sorry, can't see it. Only the old 9 element (3*3) vector.

Reply:
My bad -- I was fighting with some Windows compilation issues and overwrote my change 🤦 . Fixed up now, I swear I was trying to take your advice!! 😁

Reply:
That's alright. I was just getting a bunch of force push notifications and kept getting confused of what you changed.